### PR TITLE
SSO to populate User fullName

### PR DESCRIPTION
**Pull request by kriscfoster**

### CODE REVIEW

File: view/loadView.js [hunk 1]
```diff
@@ -108,6 +110,12 @@ function samlAuthentication(userProfile, done) {
       });
     }
 
+    if (!user.fullName) {
+      user.updateAttributes({
```
Comment (iamigo) [inline view/loadView.js]: why non-standard "updateAttributes"? shouldn't it just be "update"?

File: view/loadView.js [hunk 2]
```diff
@@ -108,6 +110,12 @@ function samlAuthentication(userProfile, done) {
       });
     }
 
+    if (!user.fullName) {
+      user.update({
```
Comment (iamigo) [inline view/loadView.js]: The "done" on line 120 is going to be called *before* the user.update call executes unless you use promise semantics i.e. update(...).then(...)

**Comment by kriscfoster** [comment]
@iamigo thanks, I think this is better now

**Comment by kriscfoster** [comment]
@iamigo thanks, I think this is better now

File: view/loadView.js [hunk 3]
```diff
@@ -110,14 +110,16 @@ function samlAuthentication(userProfile, done) {
...
+    if (user.fullName) {
+      // profile already attached - default scope applied on find
+      return done(null, user);
     }
 
-    // profile already attached - default scope applied on find
-    return done(null, user);
+    user.update({
+      fullName: userFullName,
+    }).then(() => {
```
Comment (iamigo) [inline view/loadView.js]: should be `.then(() => done(null, user));`

File: view/loadView.js [hunk 4]
```diff
@@ -108,8 +110,14 @@ function samlAuthentication(userProfile, done) {
...
-    // profile already attached - default scope applied on find
-    return done(null, user);
+    if (user.fullName) {
+      // profile already attached - default scope applied on find
+      return done(null, user);
+    }
+
+    user.update({ // user.fullName doesn't exist, update user
+      fullName: userFullName,
+    }).then(() => done(null, user));
```
Comment (iamigo) [inline view/loadView.js]: Does the "user" which is passed on in the `done(null, user)` actually include the new fullName attribute? Test?

Comment (kriscfoster) [reply view/loadView.js]: @iamigo This test was here but I updated some wording to make this clearer https://github.com/salesforce/refocus/pull/789/files#diff-3036175005cd1bfda68f7b60355102bdR129 .. The test uses the user that was created in the beforeHook (without a fullName), and then checks that the fullName is there after

**Pull request merged**